rascal-vscode-extension/src/dap/RascalDebugView.ts

```diff
@@ -0,0 +1,120 @@
+        });
+        vscode.debug.onDidStartDebugSession(_e => {
+            this.changeEmitter.fire(undefined);
+        });
+        vscode.debug.onDidTerminateDebugSession(_e => {
+            this.changeEmitter.fire(undefined);
+        });
+        vscode.debug.onDidReceiveDebugSessionCustomEvent(_e => {
+            this.changeEmitter.fire(undefined);
+        });
+        context.subscriptions.push(vscode.commands.registerCommand('rascalmpl.updateDebugView', () => {
+            this.changeEmitter.fire(undefined);
+        }));
+    }
+
+    getTreeItem(element: RascalReplNode): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        return element;
+    }
+    getChildren(element?: RascalReplNode | undefined): vscode.ProviderResult<RascalReplNode[]> {
+        if (element === undefined) {
+            return this.updateRascalDebugView();
+        }
+        return [];
+    }
+    getParent?(_element: RascalReplNode): vscode.ProviderResult<RascalReplNode> {
+        return undefined;
+    }
+    resolveTreeItem?(item: vscode.TreeItem, _element: RascalReplNode, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.TreeItem> {
+        return item;
+    }
+
+    async makeLabel(label: string, processId : number) : Promise<string | vscode.TreeItemLabel> {
+        if (await this.isActiveTerminal(processId)) {
+            return {label: label, highlights : [[0, label.length]]};
+        }
+        return label;
+    }
+
+    async updateRascalDebugView() : Promise<vscode.TreeItem[]> {
+        const result : RascalReplNode[] = [];
+        for (const terminal of vscode.window.terminals) {
+            const processId = await terminal.processId;
+            if (processId === undefined) {
+                continue;
+            }
+            if (terminal.name.includes("Rascal terminal")) {
+                const label = await this.makeLabel(terminal.name, processId);
+                const serverPort = this.rascalDebugClient.getServerPort(processId);
+                const isDebugging = serverPort !== undefined && this.rascalDebugClient.isConnectedToDebugServer(serverPort);
+                const replNode = new RascalReplNode(label, serverPort, isDebugging);
+                if (serverPort !== undefined && !isDebugging) {
+                    replNode.contextValue = "canStartDebugging";
+                }
+                result.push(replNode);
+            }
+        }
+        return result;
+    }
+
+    async isActiveTerminal(processId : number) : Promise<boolean> {
+        const activeTerminal = vscode.window.activeTerminal;
+        if (activeTerminal === undefined) {
+            return false;
+        }
+        return processId === await activeTerminal.processId;
+    }
+
+}
+
+export class RascalReplNode extends vscode.TreeItem {
+    serverPort? : number | undefined;
+
+    constructor(label : string | vscode.TreeItemLabel, serverPort : number | undefined, isDebugging : boolean) {
```

Comment:
nit: if we write `public readonly serverPort?: number | undefined` in the constructor definition, it automatically becomes an field.